src/compiler/checker.ts

```diff
@@ -12583,7 +12583,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         const baseTypes = getBaseTypes(source);
         if (baseTypes.length) {
             if (source.symbol && members === getMembersOfSymbol(source.symbol)) {
-                members = createSymbolTable(source.declaredProperties);
+                members = createSymbolTable(arrayFrom(members.values()));

```

Comment:
Huh, `createSymbolTable` probably should take `Iterable<Symbol>` given its implementation. I guess we've never given it anything but arrays.